mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -400,65 +426,68 @@ def _compute_moc_postprocess(config, streams, variableMap, calendar,
+    outputFileTseries = '{}/mocTimeSeries.nc'.format(outputDirectory)
+
+    continueOutput = os.path.exists(outputFileTseries)
+    if continueOutput:
+        # read in what we have so far
+        dsMOCTimeSeries = xr.open_dataset(outputFileTseries)
+        # force loading and then close so we can overwrite the file later
+        dsMOCTimeSeries.load()
+        dsMOCTimeSeries.close()
+    else:
+        # create a new data set from a dictionary
+        dictonary = {
+            'Time': {'dims': ('Time'),
+                     'units': 'days since 0001-01-01',
+                     'data': []},
+            'mocAtlantic26': {'dims': ('Time'),
+                              'units': 'Sv (10^6 m^3/s)',
+                              'description': 'Max MOC Atlantic streamfunction '
+                                             'nearest to RAPID Array '
+                                             'latitude (26.5N)',
+                              'data': []}}
+        dsMOCTimeSeries = xr.Dataset.from_dict(dictonary)
+
+    newEntries = False
+    for tIndex in range(ds.Time.size):
+        if ds.Time[tIndex].values in dsMOCTimeSeries.Time.values:
+            # we have it already, so no need to recompute
+            continue
+        date = days_to_datetime(ds.Time[tIndex], calendar=calendar)
+        print '   date: {:04d}-{:02d}'.format(date.year, date.month)
+        horizontalVel = ds.avgNormalVelocity[tIndex, :, :].values
+        verticalVel = ds.avgVertVelocityTop[tIndex, :, :].values
+        velArea = verticalVel * areaCell[:, np.newaxis]
+        transportZ = _compute_transport(maxEdgesInTransect,
+                                        transectEdgeGlobalIDs,
+                                        transectEdgeMaskSigns,
+                                        nVertLevels, dvEdge,
+                                        refLayerThickness,
+                                        horizontalVel)
+        mocTop = _compute_moc(latAtlantic, nVertLevels, latCell,
+                              regionCellMask, transportZ, velArea)
+        mocAtlantic26 = np.amax(mocTop[:, indlat26])
+
+        dictonary = {'Time': {'dims': ('Time'), 'data': [ds.Time[tIndex]]},
+                     'mocAtlantic26': {'dims': ('Time'),
+                                       'data': [mocAtlantic26]}}
+        # add the new time entry to the data set
+        dsMOCTimeSeries = xr.concat([dsMOCTimeSeries,
+                                     xr.Dataset.from_dict(dictonary)],
+                                    dim='Time')
+        newEntries = True
+        if date.month == 12:
+            # each year, save the data set for safe keeping
+            print 'saving progress.'
+            dsMOCTimeSeries = write_file(dsMOCTimeSeries)
+            newEntries = False
+
+    if newEntries:
+        print 'saving final result.'
+        dsMOCTimeSeries = write_file(dsMOCTimeSeries)
+
+    return dsMOCTimeSeries  # }}}
```

Comment:
@milenaveneziani, above here is the heart of the changes I made to improve file caching.  Make sure it makes sense to you and you approve of the approach.